plugins/action/synchronize.py

```diff
@@ -347,6 +347,7 @@ def run(self, tmp=None, task_vars=None):
 
                 else:
                     user = task_vars.get('ansible_user') or self._play_context.remote_user
+                user = self._templar.template(user)
```

Comment:
the vars were always the wrong thing to get, the correct info should be from the connection itself using  `get_option`

Reply:
Seems like a good suggestion. Would we need to parse (ssh_args)[https://github.com/ansible-collections/ansible.posix/blob/7ac665a0aa169c0f3f4e0b586eb5998f481001cd/plugins/action/synchronize.py#L182-L188] to get the user? Or is that exposed as an option? Do we need to consider other connection plugins other than ssh?

Reply:
ping @bcoca @jovial - this conversation is a bit stale :)

Reply:
all connections at this point should be relying on `get_option` some older ones might still be using `play_context.<data>` but this is incorrect as it does not always reflect changes do to modifiers in the task (mostly loops).